pkg/controllers/managementuser/rbac/roletemplates/roletemplate_handler.go

```diff
@@ -0,0 +1,258 @@
+	if rt.Context == projectContext {
+		promotedRules := getPromotedRules(rt.Rules)
+
+		// If there are no promoted rules and no inherited RoleTemplates, no need for additional cluster roles
+		if len(promotedRules) == 0 && len(rt.RoleTemplateNames) == 0 {
+			return rt, nil
+		}
+
+		if len(promotedRules) != 0 {
+			// 3. Project global resources cluster role
+			cr = rtl.buildClusterRole(promotedClusterRoleNameFor(rt), rt.Name, promotedRules)
+			err = rtl.createOrUpdateClusterRole(cr)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// 4. Project global resources aggregating cluster role
+		// It's possible for this role to have no rules if there are no promoted rules in any of the inherited RoleTemplates or in the above ClusterRole (3)
+		// but without fetching all those RoleTemplates and looking through their rules, it's not possible to prevent this ahead of time as the Rules in
+		// an aggregating cluster role only get populated at run time
+		cr = rtl.buildAggregatingClusterRole(promotedClusterRoleNameFor(rt), rt.Name, rt.RoleTemplateNames)
+		err = rtl.createOrUpdateClusterRole(cr)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return rt, nil
+}
+
+// createOrUpdateClusterRole creates or updates the given Cluster Role
+func (rtl *roleTemplateLifecycle) createOrUpdateClusterRole(cr *v1.ClusterRole) error {
+	// attempt to get the cluster role
+	clusterRole, err := rtl.crClient.Get(cr.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// Cluster Role doesn't exist, create it
+		_, err = rtl.crClient.Create(cr)
+		if apierrors.IsAlreadyExists(err) {
+			// in the case where it got created at the same time somehow, attempt to get it again
+			clusterRole, err = rtl.crClient.Get(cr.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+		} else if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	} else if err != nil {
+		return err
+	}
+
+	// check that the existing cluster role is the same as the one we want
+	if hasClusterRoleChanged(clusterRole, cr) {
+		// if it has changed, update it to the correct version
+		_, err := rtl.crClient.Update(cr)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// hasClusterRoleChanged returns true if the current ClusterRole differs from the expected ClusterRole
+// TODO: customize this comparison
+// reflect.DeepEqual is too strong of a check because our expected CR won't have metadata, labels applied elsewhere, etc
+func hasClusterRoleChanged(currentCR, expectedCR *v1.ClusterRole) bool {
+	return !reflect.DeepEqual(currentCR, expectedCR)
+}
+
+// Remove deletes all ClusterRoles created by the RoleTemplate
+func (rtl *roleTemplateLifecycle) Remove(rt *v3.RoleTemplate) (runtime.Object, error) {
+	var returnedErrors error
+
+	crName := clusterRoleNameFor(rt)
+	err := rtl.crClient.Delete(crName, &metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		returnedErrors = errors.Join(returnedErrors, err)
+	}
```

Comment:
I think it's worth adding a `tryDelete` that deletes and ignores NotFound errors.